src/components/widgets/edit-profile/edit-profile.component.js

```diff
@@ -0,0 +1,144 @@
+import React, { useReducer, useEffect } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import EditIcon from '@material-ui/icons/Edit';
+import PropTypes from 'prop-types';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Chip from '@material-ui/core/Chip';
+import FormControl from '@material-ui/core/FormControl';
+
+const reducer = (state, {
+  type, value, key
+}) => {
+  switch (type) {
+    case 'PROFILE_CHANGED':
+      return { ...state, ...value };
+    case 'ADD_SKILL':
+      return { ...state, skills: value ? [...state.skills, value] : [...state.skills] };
+    case 'REMOVE_SKILL':
+      return { ...state, skills: state.skills.filter(element => element !== value) };
+    case 'VALIDATE':
+      return value;
+    default:
+      return { ...state, [key]: value };
+  }
+};
+
+export default function EditProfile({ userData, handleUserDataChange }) {
+  const initialState = {
+    setOpen: false,
+  };
+  const [state, dispatch] = useReducer(reducer, initialState);
+  useEffect(() => {
+    dispatch({ value: userData, type: 'PROFILE_CHANGED' });
+  }, [userData]);
+  const handleClickOpen = () => {
+    dispatch({ value: true, key: 'setOpen' });
+  };
+
+  const handleClose = () => {
+    dispatch({ value: false, key: 'setOpen' });
+  };
+
+  const handleSubmit = () => {
+    const modifiedUserData = { ...state };
+    delete modifiedUserData.setOpen;
+    handleUserDataChange(modifiedUserData);
+    dispatch({ value: false, key: 'setOpen' });
+  };
+
+  const handleChange = ({ target }) => dispatch({ value: target.value, type: 'ADD_SKILL' });
+  return (
+    <div>
+      <Button
+        variant="contained" color="primary"
```

Comment:
Wrong styleing, if you put one prop into a new line then put all of it into a new line.